### PR TITLE
Remove unnecessary step for the image

### PR DESCRIPTION
**Pull request by abunashir**
The current version of the image is explicitly installing rvm, setting up user, then installing ruby with bundler and then install everything using bundler which should not be necessary for our use case.

This commit changes this, now it usages a ruby slim image as a base, so we don't have to set up anything for ruby explicitly. This also installs gems directly and keeps the version number as a variable.

Related: #19

### CODE REVIEW

**Comment by abunashir** [comment]
@ronaldtse, @CAMOBAP795: Could you please test this one out locally and see if that covers all of our edge cases?

**Comment by ronaldtse** [comment]
@CAMOBAP795 can you help review and test this change? Thanks!

**Comment by ronaldtse** [comment]
@abunashir the reason why we have two images `metanorma` and `mn`:
* `metanorma` is the normal `gem install metanorma-cli`
* `mn` is the edge build, occasionally we have to add or fix a flavor for users.

**Comment by abunashir** [comment]
@ronaldtse I like this approach of having two images, but just a thought what do you think about changing the metanroma version to be an environment variable, so by default, it will use the default version but if the user wants then they can also change it to any version they like?

**Comment by ronaldtse** [comment]
That’s a good idea. But I’m not sure if it’s possible to run old versions of metanorma-cli without the properly versioned dependencies...?

**Comment by abunashir** [comment]
That's a good question, I'm not sure, to be honest, don't we have all the dependency specified in each of our `metanorma-cli` version?

**Comment by ronaldtse** [comment]
@abunashir the dependencies are specified, but often using something like `>= 1.2`. It is very possible that a newer `metanorma-cli` uses versions that are not backwards compatible. I'm not confident, but we can definitely try.

**Comment by ronaldtse** [comment]
@abunashir there is also an issue where we wanted to use a Gemfile.lock to lock the versions in this image.

It allows us to guarantee all the gem versions in the image. Without that, we cannot ensure that all the gems interoperate properly. This is due to our suboptimal state of gems dependency.

Could you please revert that change? i.e. remove RVM, but keep `bundler`.

**Comment by abunashir** [comment]
@ronaldtse: I'm not sure if I'm missing something here, why do we want to have a lock file in a docker image? Normally bundler will ensure those based on the `metanorma-cli` gemspec, but we want to be really strict about it then should not that be in the gem itself?

**Comment by ronaldtse** [comment]
@abunashir without a lock file, we will not be able to generate the same docker image on every run.

**Comment by abunashir** [comment]
@ronaldtse: you mean building the image? When you are running an image isn't it just running a pre-built image with everything installed in it?

**Comment by ronaldtse** [comment]
@abunashir example:

* Today: metanorma-cli is at 1.2.2, metanorma-iso is at version X. The built image will contain metanorma-iso version X.

* Tomorrow: metanorma-cli is at 1.2.2, metanorma-iso is at version Y. The built image will contain metanorma-iso version Y, even when the Dockerfile / metanorma-cli version is not changed.

**Comment by abunashir** [comment]
@ronaldtse: From my understanding, the docker image is more likely a snapshot, once you build it and pushed it to docker hub with tag, then no matter at what point of time you run that image tag it will always be the same - it's not running your docker file again and again but just downloading the snapshot that contains all dependency pre-installed, So why would that change tomorrow?

For example, if run the `metanorma:1.1.6` image today and look at the dependency, and when it is created is the one when you actually build the image not when you ran the image

 
Screenshot 2019-07-09 14 19 21


Or do you mean, if you build using the Dockerfile tomorrow and run that image?

**Comment by ronaldtse** [comment]
@abunashir what I mean is that when you build the docker image from the same Dockerfile, there should be some certainty of what the outcome will be.

i.e. the building of the docker image should be idempotent -- if I run it three times, the three resulting images should be identical (most importantly the metanorma versions).

The Metanorma stack is a bit unstable such that old/new versions may not play nice with each other. Sometimes we have to force a dependency gem version. Do you understand what I mean?

**Comment by abunashir** [comment]
@ronaldtse: Just updated the PR, let me know if there is anything else.

File: Makefile [hunk 1]
```diff
@@ -101,7 +101,9 @@ build-$(3): $(3)/Gemfile.lock $(3)/Dockerfile
 		--label metanorma-container-version=$(1) \
 		--label metanorma-container-commit=$(CONTAINER_COMMIT) \
 		--label metanorma-container-commit-branch=$(CONTAINER_BRANCH) \
-		.
+		.;\
+
+	make clean-$(3)
```
Comment (ronaldtse) [inline Makefile]: Use `$$(MAKE) clean-$(3)` to use the same `$(MAKE)` command.

File: metanorma/Dockerfile.in [hunk 2]
```diff
@@ -2,76 +2,21 @@ FROM ${ROOT_IMAGE}
...
-
-# USER ${RVM_USER}
+# export java executable path
+ENV JAVA_HOME /usr/lib/jvm/default-java
+ENV PATH $PATH:$JAVA_HOME/bin
 
+# cleanup and set base entrypoint
 RUN apt-get clean && rm -rf /var/lib/apt/lists/*
 
-ENTRYPOINT [ "/bin/bash", "-c", "-l" ]
```
Comment (ronaldtse) [inline metanorma/Dockerfile.in]: @abunashir have you tried using arguments to the `metanorma` command, such as:
* `metanorma compile -x html blah.adoc`
* `metanorma new -t ietf -d rfc my-new-doc`
?

Just wanted to make sure it works. We could add these to the `test` Makefile target. Thanks!

Comment (abunashir) [reply metanorma/Dockerfile.in]: Yap, it seems to be working fine.

Screenshot 2019-07-15 17 00 00

Comment (ronaldtse) [reply metanorma/Dockerfile.in]: Thanks @abunashir !

**Comment by ronaldtse** [comment]
@CAMOBAP795 can you confirm the resulting image is pushed to Docker Hub? Thanks!

**Comment by andrew2net** [comment]
@ronaldtse what do you suggest me to do with the PR?

File: Makefile [hunk 3]
```diff
@@ -25,9 +25,8 @@ REPO_GIT_NAME ?= $(shell git config --get remote.origin.url)
 
 ITEMS       ?= 1 2
 IMAGE_TYPES ?= metanorma mn
-VERSIONS		?= 1.1.8 1.1.8
-ROOT_IMAGES ?= ubuntu:18.04 ubuntu:18.04
-RUBY_VER = 2.5.4
+VERSIONS		?= 1.1.8 1.2.2
```
Comment (CAMOBAP795) [inline Makefile]: I suppose both versions should be 1.2.2 here

Comment (andrew2net) [reply Makefile]: Actually, I'm out of the context. I didn't have a chance to get into `metanorma-docker` yet.

Comment (ronaldtse) [reply Makefile]: > I suppose both versions should be 1.2.2 here

Yes, @CAMOBAP795 can you make a PR? Thanks.

Comment (ronaldtse) [reply Makefile]: @andrew2net sorry I tagged you wrongly 🤣

**Comment by ronaldtse** [comment]
@CAMOBAP795 `mn` and `metanorma` are separated in order to represent:

* `metanorma`: release
* `mn`: edge

I think this PR didn't update `mn`. Can you help do it (and tag)? Thanks!

**Comment by CAMOBAP795** [comment]
@ronaldtse so `mn` should use gems from GitHub (not from rubygems.org), right?

**Comment by ronaldtse** [comment]
> @ronaldtse so `mn` should use gems from GitHub (not from rubygems.org), right?

Unfortunately not all `mn` GitHub gems are guaranteed working -- I think `mn` was only used if we wanted to override something, e.g. inclusion of new flavor.
